### ISSUE TITLE
Model selection curve for Random Input Dropout

### ISSUE THREAD

**Issue by charlesincharge** [DistrictDataLabs/yellowbrick#1204]
**Describe the solution you'd like**
A plot like https://www.scikit-yb.org/en/latest/api/model_selection/rfecv.html, but with randomly selected features (instead of importance-weighted).

**Is your feature request related to a problem? Please describe.**
In neural decoding research, we are often interested in how many recording electrodes/features are needed to achieve a given performance. This gives context to how well the variable-of-interest might be decoded in another participant (with their own electrodes). However, "feature 1" in participant A often doesn't correspond directly to "feature 1" in participant B, so we can't compare exact features, but we can, for example, compare performance of 30 electrodes in A vs 30 electrodes in B.

Perhaps this would be useful in other applications where feature importances might vary over time, etc (e.g. air quality sensors in different locations, or number of pixels in an input image).

**Examples**
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8293867/figure/F3/, panels C and H

**Potential implementation ideas**
Could either:
* Add an option to [RFECV](https://www.scikit-yb.org/en/latest/api/model_selection/rfecv.html#yellowbrick.model_selection.rfecv.RFECV) to use a different (i.e., random) feature selection metric, instead of `coef_`
* Create a pipeline with [validation_curve](https://www.scikit-yb.org/en/latest/api/model_selection/validation_curve.html) to use [sklearn's SelectKBest](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.SelectKBest.html#sklearn.feature_selection.SelectKBest) but with a random scoring metric

**Comment by bbengfort** [comment]
@charlesincharge thank you for contributing to Yellowbrick, I think this is indeed something that would be useful in a variety of applications and machine learning uses cases. Perhaps instead of adding this as an option to RFECV, we could create a new visualizer in the `model_selection` package that implements this type of feature selection? We'd be happy to review a PR, any interest in taking a crack at this for us?

**Comment by charlesincharge** [comment]
Sounds good! Might take 1-2 months before I have time to get to this, but I'm happy to give it a shot.

**Comment by charlesincharge** [comment]
Pull request opened,  #1206

**Comment by charlesincharge** [comment]
@bbengfort are you able to take a look at the pull request? I had a question there as well. Thanks!

**Comment by bbengfort** [comment]
Hi @charlesincharge Happy New Year, and thank you for opening the PR! I've checked in with some of the other YB maintainers, but if they're not available I'll try to take a look as soon as I can.

**Comment by charlesincharge** [comment]
Thank you!
